lit/SwiftREPL/SwiftInterfaceForceModuleLoadMode.test

```diff
@@ -0,0 +1,65 @@
+// This test checks that the value of the SWIFT_FORCE_MODULE_LOADING environment
+// variable is respected when loading Swift modules.
+
+// RUN: rm -rf %t && mkdir %t
+
+// Setup generates a dylib and .swiftinterface for A.swift as is, but generates
+// the .swiftmodule from a modified version a A.swift where the 'FromInterface'
+// type has been renamed to 'FromSerialized'. This means that the 'testValue'
+// variable will have a different type depending on whether the module was
+// loaded via the .swiftinterface or .swiftmodule.
+
+// RUN: mkdir %t/lib
+// RUN: cp %S/Inputs/A.swift %t/AA.swift
+// RUN: %target-swiftc -module-name AA -emit-parseable-module-interface-path %t/lib/AA.swiftinterface -emit-library -o %t/lib/libAA%target-shared-library-suffix %t/AA.swift
+// RUN: sed -e 's/FromInterface/FromSerialized/g' %t/AA.swift | %target-swiftc -module-name AA -emit-module -o %t/lib/AA.swiftmodule -
+// RUN: rm %t/AA.swift
+
+
+// This is the input provided to lldb. It intentionally assigns 'testValue' to
+// a variable of the incorrect type so we can tell via the produced diagnostic
+// whether the .swiftinterface or .swiftmodule was used.
+
+import AA
+
+let x: OtherType = testValue
+// NOT-LOADED: use of undeclared type 'OtherType'
+// FROM-INTERFACE: cannot convert value of type 'FromInterface' to specified type 'OtherType'
+// FROM-SERIALIZED: cannot convert value of type 'FromSerialized' to specified type 'OtherType'
+// INVALID: error: unknown value for SWIFT_FORCE_MODULE_LOADING variable: '{{.*}}'
+}
+
+// 1. Neither .swiftinterface or .swiftmodule present
+//
+// RUN: mv %t/lib %t/lib-backup
+// RUN: env SWIFT_FORCE_MODULE_LOADING=prefer-parseable %lldb --repl="-I%t/lib -L%t/lib -lAA -enable-parseable-module-interface" < %s 2>&1 | FileCheck %s -check-prefix=NOT-LOADED
```

Comment:
I don't think you need the `-enable-parseable-module-interface` here; the environment variable implies it for the compiler and we're just always enabling it for LLDB.